src/plugins/borrow-plugins/plugins/aave/BorrowEngineFactory.js

```diff
@@ -122,22 +144,79 @@ export const makeBorrowEngineFactory = (blueprint: BorrowEngineBlueprint) => {
       },
 
       async deposit(request: DepositRequest): Promise<ApprovableAction> {
-        return {
-          networkFee: {
-            currencyCode: wallet.currencyInfo.currencyCode,
-            nativeAmount: '2100000000000000'
-          },
-          approve: async () => {}
+        const { nativeAmount, tokenId, fromWallet } = asDepositRequest(request)
```

Comment:
This is a bad way to use cleaners. We aren't tying to go from `any` or `mixed` to typed data - we already know the types. We are just trying to validate constraints. Therefore, it should probably look more like this:

```js
const { nativeAmount, tokenId, fromWallet } = request
validateWalletParam(fromWallet)

const token = getToken(tokenId)
const tokenAddress = getContractAddress(tokenId)
```

This is the same logic you already have, but with the confusing & misleading function names changed to reflect what is actually happening.

Reply:
I took your suggestion.